src/dmd/expressionsem.d

```diff
@@ -7811,6 +7811,24 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             }
             if (!exp.lwr || !exp.upr)
             {
+                if (auto ts = tp.next.isTypeStruct())
```

Comment:
The same error applies to pointers / enum members:

```d
struct S {
	int[] opIndex() { return a; }
	int[] a;
}

// opSlice alternative
class C {
	int[] opIndex() { return a; }
	int[] a;
}

enum E : S
{
    a = S([1])
}


void main() {
	S* s;
	C* c;
	E* e;
    
	int* p;

	p = s[].ptr;
	p = c[].ptr;
    p = e[].ptr;
    
	p = (*s)[].ptr;
	p = (*c)[].ptr;
    p = (*e)[].ptr;
}
```

Reply:
Done